webserver/webserver/src/main/java/io/helidon/webserver/http1/Http1ServerResponse.java

```diff
@@ -693,9 +698,15 @@ private void sendFirstChunkOnly() {
         }
 
         private void sendHeadersAndPrepare() {
+            Status usedStatus = status.get();
+
             if (headers.contains(HeaderNames.CONTENT_LENGTH)) {
                 contentLength = headers.contentLength().orElse(-1);
                 isChunked = false;
+            } else if (isNoEntityStatus(usedStatus)) {
```

Comment:
I think this should be done in the `io.helidon.webserver.http.ServerResponse#status(io.helidon.http.Status)` only - when you set status, we should do the update of content length (unless already set). This seems too late.
There should also be a guarantee that you CANNOT send entity once you do this - i.e. if the user sets explicitly `chunked` transfer encoding, and you set `Content-Length=0`, and then the user sends an entity using an output stream, we have to provide a state that will not break the connection, and at the same time will provide a reasonable error message.

Reply:
That approach will require a number of changes in other places. The related issue is undoubtedly an edge case, where the output stream is requested before the status is set; our internal logic is currently switching to chunked encoding when the output stream is requested, and setting the status at a later time will result in other exceptions in `BlockingOutputStream`. 

@ ksivap147 How much is your logic affected if you handle the 204 case separately, and not after the output stream was requested from the response?